cmd/snapshot-controller/main.go

```diff
@@ -59,6 +63,40 @@ var (
 	version = "unknown"
 )
 
+// Checks that the VolumeSnapshot v1 CRDs exist.
+func ensureCustomResourceDefinitionsExist(client *clientset.Clientset) error {
+	condition := func() (bool, error) {
+		var err error
+		_, err = client.SnapshotV1().VolumeSnapshots("kube-system").List(context.TODO(), metav1.ListOptions{})
```

Comment:
Are we sure "kube-system" namespace exist in all Kubernetes distributions?

Reply:
From https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/#working-with-namespaces

> Kubernetes starts with four initial namespaces:
>
> - default The default namespace for objects with no other namespace
> - kube-system The namespace for objects created by the Kubernetes system
> - kube-public This namespace is created automatically and is readable by all users (including those not authenticated). This namespace is mostly reserved for cluster usage, in case that some resources should be visible and readable publicly throughout the whole cluster. The public aspect of this namespace is only a convention, not a requirement.
> - kube-node-lease This namespace for the lease objects associated with each node which improves the performance of the node heartbeats as the cluster scales.

I'm not sure if it can be deleted later though

Reply:
Can you add a check to see if "kube-system" exist before checking if there are snapshot resources there?

Reply:
Done, if kube-system doesn't exist we skip the volumesnapshot list request